kmod/src/dir.c

```diff
@@ -552,8 +551,7 @@ static int KC_DECLARE_READDIR(scoutfs_readdir, struct file *file,
 
 	dent = alloc_dirent(SCOUTFS_NAME_LEN);
 	if (!dent) {
-		ret = -ENOMEM;
-		goto out;
+		return -ENOMEM;
```

Comment:
Yeah, the lock should have been initialized to null as you noticed and fixed :).   In general, we try to have our teardown functions work for uninitialized inputs so that we can have these single-exit patterns without having to worry about how we get to them.